src/common-libs/plan/index.ts

```diff
@@ -0,0 +1,511 @@
+import { Database } from '@unocha/hpc-api-core/src/db';
+import { PlanReportingPeriodId } from '@unocha/hpc-api-core/src/db/models/planReportingPeriod';
+import { InstanceDataOfModel } from '@unocha/hpc-api-core/src/db/util/raw-model';
+import { Brand, createBrandedValue } from '@unocha/hpc-api-core/src/util/types';
+import { AddPlanTagInput } from '../../domain-services/plantag/graphql/types';
+
+export const getVersion = async (
+  models: Database,
+  planTag: AddPlanTagInput
+): Promise<string> => {
+  const latestPlanTag = await models.planTag.find({
+    where: {
+      planId: createBrandedValue(planTag.planId),
+    },
+    orderBy: { column: 'createdAt', order: 'desc' },
+  });
+  const latestTagName =
+    (latestPlanTag.length && latestPlanTag[0].name) || '0.0';
+  let [majorVersion, minorVersion] = latestTagName
+    .split('.')
+    .map((v) => parseInt(v) || 0);
```

Comment:
Let's use a radix, just to be safe
```suggestion
    .map((v) => parseInt(v, 10) || 0);
```